### ISSUE TITLE
Tensorflow 1.13 - can't deprecate tf.layers.batch_normalization.  No existing replacement

### ISSUE THREAD

**Issue by jtressle** [tensorflow/tensorflow#26146]
I'm updating to Tensorflow 1.13 from 1.12.

tf.layers.batch_normalization is being deprecated.

The note states:

```
batch_normalization (from tensorflow.python.layers.normalization) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.batch_normalization instead.
```
First, there is no 'keras.layers.batch_normalization'. This is a 'keras.layers. BatchNormalization', but that is a different function and a direct replacement for 'tf.layers. BatchNormalization'.

Therefore, either make a 'keras.layers. BatchNormalization' function or remove the deprecation warning for 'tf.layers.batch_normalization.

Here's the link to the batch_normalization fuction.

https://github.com/tensorflow/tensorflow/blob/r1.12/tensorflow/python/layers/normalization.py

Thanks,

**Comment by MarkDaoust** [comment]
We are not planning to support these function-based layers in the future, as they rely on hidden global state. After tensor 2.0 there will only be object oriented layers.

> keras.layers.batch_normalization

That is a typo. 

Can you send us a PR to fix it to `keras.layers.BatchNormalization`? It's easy with the online editor under the pencil button in the top right:
 
This line:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/layers/normalization.py#L160

**Comment by jtressle** [comment]
@MarkDaoust Understood and I submitted a PR.  

Will there be a migration tutorial or examples on how to convert from TF 1.X to 2.0?  

I have a custom model trained using batch_normalization in Tensorflow 1.2.  If I want to use its checkpoint in TF 2.0, what would the steps be to use that model?  I'm worried TF would not recognize any variables that maybe affected by the different batch_normalizations.

Thanks!

**Comment by MarkDaoust** [comment]
Thanks @jtressle.

> Will there be a migration tutorial or examples on how to convert from TF 1.X to 2.0?

Yes you can preview the docs at: https://github.com/tensroflow/docs/site/en/r2
There is a guide to the update script, and one checkpointing doc,  but I'm not sure it has those specific details.

https://github.com/tensorflow/docs/blob/master/site/en/r2/guide/upgrade.md
https://github.com/tensorflow/docs/blob/master/site/en/r2/guide/checkpoints.ipynb

There will be a more in depth, practical guide to updating, coming soon.

**Comment by MarkDaoust** [comment]
And don't forget to send a pull request. Right now that commit is just sitting in your fork.

**Comment by jtressle** [comment]
@MarkDaoust Pull request 26154 has been submitted.  Yeah I forgot to send in the pull.

Thanks

**Comment by amahendrakar** [comment]
@jtressle,
Closing this issue as PR [#26154](https://github.com/tensorflow/tensorflow/pull/26154) has been merged. Please feel free to re-open if mistaken. Thanks!
